.github/workflows/python.yaml

```diff
@@ -0,0 +1,46 @@
+name: ci-cd-pipeline
+
+on: [push, pull_request]
+jobs:
+  test:
+    runs-on: ubuntu-20.04
+    container: python:3.8-slim
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install dependencies
+        run: |
+          pip install poetry
+          poetry install --extras "all"
+      - name: Formatting
+        run: |
+          poetry run black --check .
+      - name: Sorted Imports
+        run: |
+          poetry run isort --check-only --profile black --line-length 88 .
+      - name: Type Checks
+        continue-on-error: true
+        run: poetry run mypy --config-file pyproject.toml .
+      - name: Linting
+        continue-on-error: true
+        run: |
+          poetry run pylint --errors-only parameterspace/
+          poetry run pylint --errors-only tests/
+      - name: Run tests
+        if: always()
```

Comment:
Would you be so kind to elaborate a bit on what this condition does for us?

Reply:
The thing I don't really like in the pipeline yet is that you get the test results slice by slice: If you didn't auto-format correctly, the test breaks, you have to push a formatted fix and then find out that you also forgot isort and so on until the tests.

Here I wanted to at least always run the functional tests, as fixing those usually requires changes that impact linter/auto formatting.

Still, it's not very concise and I'm not 100% satisfied with it.

Do you have any thoughts or ideas on that?

Reply:
As discussed in person, we'll just try to add `if: always()` to any test and see how it feels.